### PR TITLE
152: Shows ranked explanation on game creation

### LINKED ISSUE #152: [Feature]: Show explanation of Ranked play in `CreateGameDialog`
### Feature Summary

When user's create a game, they see a switch to toggle whether the game will be 'Ranked'. We should provide additional explanation for new players so they can understand what it means for a game to be Ranked.

### Detailed Description

See pic of `CreateGameDialog` below:
image

We could and should reuse the existing `StatsScoringDIalog` to explain the ranked scoring in the `CreateGameDialog`. See pics below for its usage in the `Stats` page:
image

The opened dialog
image

Currently that dialog refers to 'this page' in reference to the stats page. This language should be generalized so as to apply equally in the `CreateGameDialog`

### PR DESCRIPTION
**Pull request by byarmis**
## Please check the following

- [x] Do the tests still pass? (see [Run the Tests](https://github.com/cuttle-cards/cuttle#run-the-tests))
- [x] Is the code formatted properly? (see [Linting (Formatting)](https://github.com/cuttle-cards/cuttle#linting-formatting))
- For New Features:
  - [x] Have tests been added to cover any new features or fixes?
  - [x] Has the documentation been updated accordingly?

## Please describe additional details for testing this change

image

image

Resolves #152

### CODE REVIEW

File: client/js/components/CreateGameDialog.vue [hunk 1]
```diff
@@ -17,7 +17,14 @@
...
       </v-card-title>
       <v-card-text>
-        <v-switch v-model="isRanked" label="Ranked" data-cy="create-game-ranked-switch" />
+        <div class="d-flex align-center">
+          <div>
+            <v-switch v-model="isRanked" label="Ranked" data-cy="create-game-ranked-switch" />
+          </div>
+          <div>
+            <stats-scoring-dialog :show-text="false" data-cy="ranked-info-button" />
+          </div>
```
Comment (itsalaidbacklife) [inline client/js/components/CreateGameDialog.vue:26]: Any reason for the wrapping `div`s?
```suggestion
          <v-switch v-model="isRanked" label="Ranked" data-cy="create-game-ranked-switch" />
          <stats-scoring-dialog :show-text="false" data-cy="ranked-info-button" />
```

File: client/js/components/CreateGameDialog.vue [hunk 2]
```diff
@@ -55,8 +62,11 @@
 <script>
 const LS_PREFERS_RANKED_NAME = 'prefersRanked';
 
+import StatsScoringDialog from '@/components/StatsScoringDialog.vue';
```
Comment (itsalaidbacklife) [inline client/js/components/CreateGameDialog.vue:65]: prefer imports above constant declaration (move this above line 63)

File: client/js/components/StatsScoringDialog.vue [hunk 3]
```diff
@@ -50,9 +50,10 @@
           </v-list-item>
         </v-list>
         <p>
-          You can view the statistics and rankings for each season on this page. At the end of each
-          season, the top 4 players compete in a double elimination championship tournament. Do you
-          have what it takes to become
+          You can view the statistics and rankings for each season on
+          <a href="#/stats" target="_self"> the Stats page </a> . At the end of each season, the top
```
Comment (itsalaidbacklife) [inline client/js/components/StatsScoringDialog.vue:54]: Does this work nicely? Normally we use vue's `<router-link>` component for internal SPA routing. @seriouslysean got an opinion on this?

File: client/js/components/StatsScoringDialog.vue [hunk 4]
```diff
@@ -72,6 +73,12 @@ export default {
   components: {
     AwardCard,
   },
+  props: {
+    showText: {
```
Comment (itsalaidbacklife) [inline client/js/components/StatsScoringDialog.vue:77]: Might be more clear as `showButtonText`

**Review by itsalaidbacklife — APPROVED** [review]
Looks good. Questions & comments submitted

File: client/js/components/CreateGameDialog.vue (hunk 1 above)
Comment (byarmis) [reply client/js/components/CreateGameDialog.vue:26]: huh, thought I tried that and they ended up getting stacked vertically.  Guess not!

File: client/js/components/StatsScoringDialog.vue (hunk 3 above)
Comment (byarmis) [reply client/js/components/StatsScoringDialog.vue:54]: Yep, think so!

https://user-images.githubusercontent.com/1594170/195737801-3adaaf64-da0d-4bc2-b2f3-cb55146e62e6.mp4

I can change it if you'd like

Comment (seriouslysean) [reply client/js/components/StatsScoringDialog.vue:54]: I would switch this. It'll work with the hash router but would force a hard reload if we ever switch to the web router, https://router.vuejs.org/guide/essentials/history-mode.html#hash-mode.

Comment (itsalaidbacklife) [reply client/js/components/StatsScoringDialog.vue:54]: Yeah seconded. @byarmis can we update this to a `<router-link>`that uses the route `name` to navigate to the `Stats` page? That'll be more flexible if we change the router mode or the page url

**Review by itsalaidbacklife — APPROVED** [review]
Looks good. Just swap the link to a `<router-link>` and we can merge

File: client/js/components/StatsScoringDialog.vue (hunk 3 above)
Comment (byarmis) [reply client/js/components/StatsScoringDialog.vue]: Sure, no problem.

Done!

**Review by itsalaidbacklife — APPROVED** [review]
Great work

**Pull request merged**